v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -166,29 +135,91 @@ export async function main(cliArguments: string[]) {
 
     console.log("Error running the task:", error.message);
 
-    if (showStackTraces) {
+    if (hhGlobalParams.showStackTraces) {
       console.log("");
       console.error(error);
     }
   }
 }
 
-function parseGlobalArguments(
-  _globalParamsIndex: GlobalParameterMap,
-  _cliArguments: string[],
-  _usedCliArguments: boolean[],
-): Partial<GlobalArguments> {
-  // TODO: Parse the global arguments
-  // - Parse the global params, skipping the processed entries
-  // - At each stage validate the value according to its type
-  // - If a global param is boolean and its default is false, its value is not
-  //   necessary. That means that if the immediatly next value is `"true"` or
-  //   `"false"`, we should use it, but if it's not we just don't, and consider
-  //   its value `true`.
-  // - Mark all the used entries in usedCliArguments
-  // - Do not resolve defaults here, the HRE does that.
-  // - Return the values
-  return {};
+export async function parseInitialHardhatParameters(
+  cliArguments: string[],
+  usedCliArguments: boolean[],
+) {
+  let configPath: string | undefined;
+  let showStackTraces: boolean = false; // true if ci
```

Comment:
@alcuadrado should we use the hh-core function [isRunningOnCiServer](https://github.com/NomicFoundation/hardhat/blob/f579adca83f0a4c47bf29cd5df9ebc9347443039/packages/hardhat-core/src/internal/util/ci-detection.ts#L20)

Reply:
That's a good point. I added a new issue for this: https://github.com/NomicFoundation/hardhat/issues/5337

I think we may reuse it from plugins/libs, so let's place it in `hardhat-utils`.

Reply:
I created [this PR](https://github.com/NomicFoundation/hardhat/pull/5340) to address this, it should be merged before this one.

Reply:
Done, the CI function is now included